R/maic_anchored.R

```diff
@@ -409,31 +492,56 @@ maic_anchored_binary <- function(res,
   res_AC$ci_u <- bin_robust_ci[2, "97.5 %"]
   res_AC$pval <- bin_robust_coef[2, "Pr(>|z|)"]
 
+  # unadjusted AC
+  res_AC_unadj <- res_template
+  res_AC_unadj$est <- summary(binobj_ipd)$coefficients[2, "Estimate"]
+  res_AC_unadj$se <- summary(binobj_ipd)$coefficients[2, "Std. Error"]
+  res_AC_unadj$ci_l <- confint.default(binobj_ipd)[2, "2.5 %"]
+  res_AC_unadj$ci_u <- confint.default(binobj_ipd)[2, "97.5 %"]
+  res_AC_unadj$pval <- summary(binobj_ipd)$coefficients[2, "Pr(>|z|)"]
+
+  # BC
   if (is.null(res_BC)) {
-    res_BC <- list(est = coef(binobj_agd)[2], se = sqrt(vcov(binobj_agd)[2, 2]))
+    res_BC <- res_template
+    res_BC$est <- summary(binobj_agd)$coefficients[2, "Estimate"]
+    res_BC$se <- summary(binobj_agd)$coefficients[2, "Std. Error"]
+    res_BC$ci_l <- confint.default(binobj_agd)[2, "2.5 %"]
+    res_BC$ci_u <- confint.default(binobj_agd)[2, "97.5 %"]
+    res_BC$pval <- summary(binobj_agd)$coefficients[2, "Pr(>|z|)"]
   }
+
+  # derive AB
   res_AB <- bucher(res_AC, res_BC, conf_lv = 0.95)
+  res_AB_unadj <- bucher(res_AC_unadj, res_BC, conf_lv = 0.95)
+
+  # transform
+  transform_ratio <- function(resobj) {
+    resobj$est <- exp(resobj$est)
+    resobj$se <- sqrt((exp(resobj$se^2) - 1) * exp(2 * resobj$est + resobj$se^2)) # log normal parameterization
```

Comment:
I think this is wrong because it is using the transformed `est` already. I don't think we use this in any calculation though.

Reply:
res_AB is an result returned from bucher function, where the default parameter for exponentiate is equal to false. So, this seems to be in log-scale
Similarly, res_AC from glm is in log-scale, so we do need to exponentiate.

I think..?

Reply:
```r
# Suppose this is the input results
resobj <- list(est = 2, se = 1, ci_l = 2 - 1.96, ci_u = 2 + 1.96)

# Then in the function:
resobj$est <- exp(resobj$est) # est = 7.389
resobj$se <- sqrt((exp(resobj$se^2) - 1) * exp(2 * resobj$est + resobj$se^2)) # se = 3497.2
```
The new se seems a bit large

Reply:
Ah I see what is wrong now..
If we flip the two rows, it should work..

resobj$se <- sqrt((exp(resobj$se^2) - 1) * exp(2 * resobj$est + resobj$se^2)) # log normal parameterization
resobj$est <- exp(resobj$est)

Or something like this.. (your preference)

transform_ratio <- function(resobj) {
    resobj2 <- list()
    resobj2$est <- exp(resobj$est)
    resobj2$se <- sqrt((exp(resobj$se^2) - 1) * exp(2 * resobj$est + resobj$se^2)) # log normal parameterization
    resobj2$ci_l <- exp(resobj$ci_l)
    resobj2$ci_u <- exp(resobj$ci_u)
    resobj2
}

Thanks for noticing. We need to change this for the unanchored binary case too!!